playwright/_impl/_connection.py

```diff
@@ -55,63 +57,69 @@ def __init__(self, connection: "Connection", object: "ChannelOwner") -> None:
         self._guid = object._guid
         self._object = object
         self.on("error", lambda exc: self._connection._on_event_listener_error(exc))
-        self._timeout_calculator: Optional[Callable[[Optional[float]], float]] = None
+        self._timeout_calculator: TimeoutCalculator = None
 
     async def send(
         self,
         method: str,
+        timeout_calculator: TimeoutCalculator,
         params: Dict = None,
         is_internal: bool = False,
         title: str = None,
     ) -> Any:
         return await self._connection.wrap_api_call(
-            lambda: self._inner_send(method, params, False),
+            lambda: self._inner_send(method, timeout_calculator, params, False),
             is_internal,
             title,
         )
 
     async def send_return_as_dict(
         self,
         method: str,
+        timeout_calculator: TimeoutCalculator,
         params: Dict = None,
         is_internal: bool = False,
         title: str = None,
     ) -> Any:
         return await self._connection.wrap_api_call(
-            lambda: self._inner_send(method, params, True),
+            lambda: self._inner_send(method, timeout_calculator, params, True),
             is_internal,
             title,
         )
 
     def send_no_reply(
         self,
         method: str,
+        timeout_calculator: TimeoutCalculator,
         params: Dict = None,
         is_internal: bool = False,
         title: str = None,
     ) -> None:
         # No reply messages are used to e.g. waitForEventInfo(after).
         self._connection.wrap_api_call_sync(
             lambda: self._connection._send_message_to_server(
-                self._object, method, {} if params is None else params, True
+                self._object,
+                method,
+                _augment_params(params, timeout_calculator, filter=False),
```

Comment:
I really don't like how this is different. It's unclear to my why `send_no_reply` implies no removal of `None`s.